yarn_api_client/main.py

```diff
@@ -145,113 +152,46 @@ def populate_history_server_arguments(subparsers):
-    hts_parser.set_defaults(method_kwargs=['type'])
-
-    ht_parser = hs_subparsers.add_parser(
-        'task', help='Task API')
-    ht_parser.add_argument('job_id')
-    ht_parser.add_argument('task_id')
-    ht_parser.set_defaults(method='job_task')
-    ht_parser.set_defaults(method_args=['job_id', 'task_id'])
-
-    htc_parser = hs_subparsers.add_parser(
-        'task_counters', help='Task Counters API')
-    htc_parser.add_argument('job_id')
-    htc_parser.add_argument('task_id')
-    htc_parser.set_defaults(method='task_counters')
-    htc_parser.set_defaults(method_args=['job_id', 'task_id'])
-
-    htas_parser = hs_subparsers.add_parser(
-        'task_attempts', help='Task Attempts API')
-    htas_parser.add_argument('job_id')
-    htas_parser.add_argument('task_id')
-    htas_parser.set_defaults(method='task_attempts')
-    htas_parser.set_defaults(method_args=['job_id', 'task_id'])
-
-    hta_parser = hs_subparsers.add_parser(
-        'task_attempt', help='Task Attempt API')
-    hta_parser.add_argument('job_id')
-    hta_parser.add_argument('task_id')
-    hta_parser.add_argument('attempt_id')
-    hta_parser.set_defaults(method='task_attempt')
-    hta_parser.set_defaults(method_args=['job_id', 'task_id', 'attempt_id'])
-
-    htac_parser = hs_subparsers.add_parser(
-        'task_attempt_counters', help='Task Attempt Counters API')
-    htac_parser.add_argument('job_id')
-    htac_parser.add_argument('task_id')
-    htac_parser.add_argument('attempt_id')
-    htac_parser.set_defaults(method='task_attempt_counters')
-    htac_parser.set_defaults(method_args=['job_id', 'task_id', 'attempt_id'])
+    listing_of_apis = [
+        'application_information',
+        'jobs',
+        'job',
+        'job_attempts',
+        'job_counters',
+        'job_conf',
+        'job_tasks',
+        'job_task',
+        'task_counters',
+        'task_attempts',
+        'task_attempt',
+        'task_attempt_counters'
+    ]
+
+    create_parsers(hs_subparsers, HistoryServer, "History Server", listing_of_apis)
 
 
 def main():
     parser = get_parser()
     opts = parser.parse_args()
 
     class_kwargs = {}
-    if not hasattr(opts, 'api_class'):
-        raise Exception("Please provide api class - rm, hs, nm, am")
     # Only ResourceManager supports HA
-    elif opts.endpoint:
+    if opts.endpoint:
         if opts.api_class == ResourceManager:
             class_kwargs['service_endpoints'] = opts.endpoint.split(",")
         else:
             class_kwargs['service_endpoint'] = opts.endpoint
 
+    # CLI requires some special accommodation for Auth - custom class imports
+    if opts.auth:
+        # Currenly only hadoop's SimpleAuth and none are supported out of the box
+        if opts.auth == 'simple':
+            class_kwargs['auth'] = SimpleAuth()
+        else:
+            raise Exception(
+                "This auth mentod is not supported by CLI, please write your own python script if needed"
```

Comment:
```suggestion
                f"This auth method ({opts.auth}) is not supported by the CLI."
```
I think leaving this short and to the point might be best.